Adding k_factor.py in dp folder in add_K_FactorofStrings branch

TheCodeYoda opened the pull request:
I have implemented an algorithm to solve a dp problem named "K-factor" 
Problem statement: The K factor of a string is defined as the number of times ‘abba’ appears as a substring.
Given two numbers N and k,​ find the number of strings of length N with ‘K factor’ = k.

ericklarac commented:
@TheCodeYoda Could you add some test cases? Also, add your implementation under the dp section on the readme file.

Use [this PR](https://github.com/keon/algorithms/pull/736/files) as a guide.

TheCodeYoda commented:
> @TheCodeYoda Could you add some test cases? Also, add your implementation under the dp section on the readme file.
> 
> Use [this PR](https://github.com/keon/algorithms/pull/736/files) as a guide.

Should I write a full tester file or just add some test cases in the program itself ?

ericklarac commented:
> > @TheCodeYoda Could you add some test cases? Also, add your implementation under the dp section on the readme file.
> > Use [this PR](https://github.com/keon/algorithms/pull/736/files) as a guide.
> 
> Should I write a full tester file or just add some test cases in the program itself ?

Add it under the `tests/test_dp.py`file, just create a class e.g. `TestKFactorOfStrings`

TheCodeYoda commented:
> > > @TheCodeYoda Could you add some test cases? Also, add your implementation under the dp section on the readme file.
> > > Use [this PR](https://github.com/keon/algorithms/pull/736/files) as a guide.
> > 
> > 
> > Should I write a full tester file or just add some test cases in the program itself ?
> 
> Add it under the `tests/test_dp.py`file, just create a class e.g. `TestKFactorOfStrings`

I have added a class in test_dp file and updated the readme but if I run the test_dp file it says "cannot import name 'find_k_factor' from 'algorithms.dp' " This is because it has not yet been merged right?

ericklarac commented:
> > > > @TheCodeYoda Could you add some test cases? Also, add your implementation under the dp section on the readme file.
> > > > Use [this PR](https://github.com/keon/algorithms/pull/736/files) as a guide.
> > > 
> > > 
> > > Should I write a full tester file or just add some test cases in the program itself ?
> > 
> > 
> > Add it under the `tests/test_dp.py`file, just create a class e.g. `TestKFactorOfStrings`
> 
> I have added a class in test_dp file and updated the readme but if I run the test_dp file it says "cannot import name 'find_k_factor' from 'algorithms.dp' " This is because it has not yet been merged right?

I think you forgot to export your algorithm, edit the `algorithms/algorithms/dp/__init__.py` file adding this line `from .k_factor import find_k_factor`. In the case you have to export more than one function, you can use `from .k_factor import *`

TheCodeYoda commented:
Even after adding from .k_factor import * in __init__.py in dp folder it seems to not detect the module find_k_factor

ericklarac commented:
> Even after adding from .k_factor import * in **init**.py in dp folder it seems to not detect the module find_k_factor

I do not see the __init__ file updated, could you try it?

TheCodeYoda commented:
Oh..ok cool Thanks
